src/VS.Web.CG.Mvc/Identity/IdentityGeneratorTemplateModelBuilder.cs

```diff
@@ -152,15 +175,142 @@ internal Type UserType
+
+            templateModel.FilesToGenerate = filesToGenerate.ToArray();
+
+            ValidateFilesToGenerate(templateModel.FilesToGenerate);
+
             return templateModel;
         }
 
-        private string GetDfaultDbContextName()
+        private void ValidateFilesToGenerate(IdentityGeneratorFile[] filesToGenerate)
+        {
+            var rootPath = _applicationInfo.ApplicationBasePath;
+            var filesToOverWrite = filesToGenerate
+                .Where(f => f.ShouldOverWrite == OverWriteCondition.WithForce
+                                && _fileSystem.FileExists(Path.Combine(rootPath, f.OutputPath)));
+
+            if (filesToOverWrite.Any() && !_commandlineModel.Force)
+            {
+                var msg = string.Format(
+                        MessageStrings.UseForceOption,
+                        string.Join(Environment.NewLine, filesToOverWrite.Select(f => f.OutputPath)));
+                throw new InvalidOperationException(msg);
+            }
+        }
+
+        private void ValidateDefaultUIOption()
+        {
+            var errorStrings = new List<string>();
+
+            if (IsFilesSpecified)
+            {
+                errorStrings.Add(string.Format(MessageStrings.InvalidOptionCombination,"--files", "--useDefaultUI"));
+            }
+
+            if (IsUsingExistingDbContext)
+            {
+                errorStrings.Add(MessageStrings.ExistingDbContextCannotBeUsedForDefaultUI);
+            }
+
+            if (errorStrings.Any())
+            {
+                throw new InvalidOperationException(string.Join(Environment.NewLine, errorStrings));
+            }
+        }
+
+        private async Task ValidateFilesOption()
+        {
+            var errors = new List<string>();
+            if (!IsUsingExistingDbContext && IsDbContextSpecified)
+            {
+                // We need the user to specify that there is an existing DbContext that inherits from IdentityDbContext.
+                errors.Add("Using '--files' option requires specifying an existing DbContext which inherits from IdentityDbContext.");
+            }
+            else if (!IsDbContextSpecified)
+            {
+                // Try to find an existing DbContext in the project (not in the dependencies) that inherits from IdentityDbContext.
+                // If 0 or more than 1 are found, ask the user to specify.
+                // If exactly one is found, use that.
+
+                var compilation = await _workspace.CurrentSolution.Projects
+                    .Where(p => p.AssemblyName == _projectContext.AssemblyName)
+                    .First()
+                    .GetCompilationAsync();
+
+                var reflectedTypesProvider = new ReflectedTypesProvider(
+                    compilation,
+                    null,
+                    _projectContext,
+                    _loader,
+                    _logger);
+
+                if (reflectedTypesProvider.GetCompilationErrors()!= null
```

Comment:
Nit: this is calling reflectedTypesProvider.GetCompliationErrors() up to 3 times (including the call in the throw). Might be more efficient to call it once, storing the result, and then checking / operating on the result.